internal/batch/batch_processor.go

```diff
@@ -98,37 +108,55 @@ func (bp *batchProcessor) assemblyLoop() {
 	defer close(bp.sealBatch) // close persitenceLoop when we exit
 	l := log.L(bp.ctx)
 	var batchSize uint
+	var batchPayloadEstimate = batchSizeEstimateBase
 	var lastBatchSealed = time.Now()
 	var quiescing bool
+	var overflowedWork *batchWork
 	for {
-		// We timeout waiting at the point we think we're ready for disposal,
-		// unless we've started a batch in which case we wait for what's left
-		// of the batch timeout
-		timeToWait := bp.conf.DisposeTimeout
-		if quiescing {
-			timeToWait = 100 * time.Millisecond
-		} else if batchSize > 0 {
-			timeToWait = bp.conf.BatchTimeout - time.Since(lastBatchSealed)
-		}
-		timeout := time.NewTimer(timeToWait)
-
-		// Wait for work, the timeout, or close
 		var timedOut, closed bool
-		select {
-		case <-timeout.C:
-			timedOut = true
-		case work, ok := <-bp.newWork:
-			if ok && !work.abandoned {
-				batchSize++
-				bp.persistWork <- work
-			} else {
-				closed = true
+		if overflowedWork != nil {
+			// We overflowed the size cap when we took this message out the newWork
+			// queue last time round the lop
+			bp.persistWork <- overflowedWork
+			batchSize++
+			batchPayloadEstimate += overflowedWork.estimateSize()
```

Comment:
Yes - I put in a bunch of work to reject this at the message send time. This is why I had to add `value_size` to the data, because we can send a `Message` that refers to an array of existing data in the system.

Reply:
https://github.com/hyperledger/firefly/blob/4478b71a89376a918e5caa7f16c99dc9e5b67a7d/internal/privatemessaging/message.go#L120-L123

Similar code in broadcast.